controllers/chat/ChatController.js

```diff
@@ -13,9 +13,133 @@ const BetterSocialConstantListUtils = require('../../services/bettersocial/const
+    let previous_channel = await ChatAnonUserInfo.findOne({
+      where: {
+        channel_id: oldChannelId,
+        my_anon_user_id: member
+      }
+    });
+    anonInfo = {
+      anon_user_info_color_code: previous_channel.anon_user_info_color_code,
+      anon_user_info_color_name: previous_channel.anon_user_info_color_name,
+      anon_user_info_emoji_code: previous_channel.anon_user_info_emoji_code,
+      anon_user_info_emoji_name: previous_channel.anon_user_info_emoji_name
+    };
+  } else {
+    anonInfo = generate_anon_user_info();
+  }
+  return anonInfo;
+};
+
+const handle_anon_to_anon_channel_owner = async (owner, member, context = null) => {
+  // check if channel already exist
+  let channel = await ChatAnonUserInfo.findOne({
+    where: {
+      my_anon_user_id: owner,
+      target_user_id: member,
+      context: context,
+      initiator: owner
+    }
+  });
+  if (channel) {
+    return channel;
+  } else {
+    // handle to create new channel if not exist
+    // generate channel id
+    let channel_id = generate_channel_id_for_anon_chat(owner, member);
+    console.log('DEBUG channel_id => ', channel_id);
+    const emoji = BetterSocialConstantListUtils.getRandomEmoji();
+    const color = BetterSocialConstantListUtils.getRandomColor();
+    let new_channel = await ChatAnonUserInfo.create({
+      channel_id: channel_id,
+      my_anon_user_id: owner,
+      target_user_id: member,
+      anon_user_info_color_code: color.code,
+      anon_user_info_color_name: color.color,
+      anon_user_info_emoji_code: emoji.emoji,
+      anon_user_info_emoji_name: emoji.name,
+      context: context,
+      initiator: owner
+    });
+    return new_channel;
+  }
+};
+
+const handle_anon_to_anon_channel_member = async (
+  channel_owner,
+  member,
+  oldChannelId,
+  context = null
+) => {
+  console.log('DEBUG channel_owner => ', channel_owner);
+  console.log('DEBUG oldChannelId => ', oldChannelId);
+  // check if channel already exist
+  let channel = await ChatAnonUserInfo.findOne({
+    where: {
+      my_anon_user_id: member,
+      target_user_id: channel_owner.my_anon_user_id,
+      context: context,
+      initiator: channel_owner.my_anon_user_id
+    }
+  });
+  if (channel) {
+    return channel;
+  } else {
+    // get anon info from old channel or generate a new one
+    let anonInfo = await get_anon_info_from_old_channel(oldChannelId, member);
+    let new_channel = await ChatAnonUserInfo.create({
+      channel_id: channel_owner.channel_id,
+      my_anon_user_id: member,
+      target_user_id: channel_owner.my_anon_user_id,
+      anon_user_info_color_code: anonInfo.anon_user_info_color_code,
+      anon_user_info_color_name: anonInfo.anon_user_info_color_name,
+      anon_user_info_emoji_code: anonInfo.anon_user_info_emoji_code,
+      anon_user_info_emoji_name: anonInfo.anon_user_info_emoji_name,
+      context: context,
+      initiator: channel_owner.my_anon_user_id
+    });
+    return new_channel;
+  }
+};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This function handles the creation of a new anonymous-to-anonymous chat channel for the member. It first checks if a channel already exists between the member and the owner's anonymous user ID with the same context and initiator. If such a channel exists, it returns that channel; otherwise, it creates a new channel with the owner's channel ID and anonymous user info from the old channel or newly generated info.
```javascript
const handle_anon_to_anon_channel_member = async (
  channel_owner,
  member,
  oldChannelId,
  context = null
) => {
  let channel = await ChatAnonUserInfo.findOne({
    where: {
      my_anon_user_id: member,
      target_user_id: channel_owner.my_anon_user_id,
      context: context,
      initiator: channel_owner.my_anon_user_id
    }
  });
  if (channel) {
    return channel;
  } else {
    let anonInfo = await get_anon_info_from_old_channel(oldChannelId, member);
    let new_channel = await ChatAnonUserInfo.create({
      channel_id: channel_owner.channel_id,
      my_anon_user_id: member,
      target_user_id: channel_owner.my_anon_user_id,
      anon_user_info_color_code: anonInfo.anon_user_info_color_code,
      anon_user_info_color_name: anonInfo.anon_user_info_color_name,
      anon_user_info_emoji_code: anonInfo.anon_user_info_emoji_code,
      anon_user_info_emoji_name: anonInfo.anon_user_info_emoji_name,
      context: context,
      initiator: channel_owner.my_anon_user_id
    });
    return new_channel;
  }
};
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->